shopping_mall/backend/app/routers/chatbot.py

```
@@ -1,36 +1,86 @@
+
+
+def _get_user_id(x_user_id: int = Header(alias="X-User-Id")) -> int:
+    return x_user_id
 
 
 @router.post("/ask", response_model=ChatAnswer)
-async def ask_question(body: ChatQuestion, db: Session = Depends(get_db)):
+async def ask_question(
+    body: ChatQuestion,
+    authenticated_user_id: int = Depends(_get_user_id),
+    db: Session = Depends(get_db)
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)